### ISSUE TITLE
get_guilds should have optional pagination

### ISSUE THREAD

**Issue by tqwewe** [serenity-rs/serenity#1375]
The get_guilds function has the following signature:
```rust
async fn get_guilds(&self, target: &GuildPagination, limit: u64) -> Result<Vec<GuildInfo>>
```
And according to the [Discord docs for guilds](https://discord.com/developers/docs/resources/user#get-current-user-guilds), the `before` and `after` fields are optional, but the Serenity Rust API forces me to pass one of them, but not none.

The API should be updated to:
```rust
async fn get_guilds(&self, target: Option<&GuildPagination>, limit: u64) -> Result<Vec<GuildInfo>>
```
Allowing an `before` and `after` to be optional.

**Comment by tylerd008** [comment]
Mind if I take this? Looks pretty straightforward and shouldn't take much time at all to do.

**Comment by drklee3** [comment]
Implemented in #1403
